### PR TITLE
[BACKLOG-40861] Prevent typing invalid connection name characters in `ConnectionDialog`

### PR DESCRIPTION
**Pull request by dcleao**
/cc @pentaho/hoth

To be merged with: https://github.com/pentaho/pentaho-scheduler-plugin-ee/pull/255

### CODE REVIEW

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

File: core/src/test/java/org/pentaho/di/connections/vfs/provider/ConnectionFileNameParserTest.java [hunk 1]
```diff
@@ -270,4 +271,48 @@ protected String shuffleString( String characterSet ) {
...
+  // region isValidConnectionNameCharacter
+  @Test
+  public void testIsValidConnectionNameCharacterReturnsFalseForInvalidCharacters() {
+    for ( char c : CONNECTION_NAME_INVALID_CHARACTERS.toCharArray() ) {
+      assertFalse( fileNameParser.isValidConnectionNameCharacter( c ) );
+    }
+  }
+
+  @Test
+  public void testIsValidConnectionNameCharacterReturnsTrueOnValidCharacters() throws FileSystemException {
```
Comment (NJtwentyone) [inline core/src/test/java/org/pentaho/di/connections/vfs/provider/ConnectionFileNameParserTest.java:284]: you have a code smell here: https://sonar.orl.eng.hitachivantara.com/project/issues?pullRequest=9455&resolved=false&types=CODE_SMELL&open=AZBzRVUFpCtKek2RFu0W&id=org.pentaho.di%3Apdi

**Review by NJtwentyone — APPROVED** [review]
conditionally approving.
Just 1 sonarqube code smell - https://github.com/pentaho/pentaho-kettle/pull/9455/files#r1662948879

I'm the first to approve, so the second approver can verify the sonarqube finding.

No need to verify back with me, when this is resolved. It can be independently verified by the absence of the sonarqube finding.

File: core/src/test/java/org/pentaho/di/connections/vfs/provider/ConnectionFileNameParserTest.java (hunk 1 above)
Comment (dcleao) [reply core/src/test/java/org/pentaho/di/connections/vfs/provider/ConnectionFileNameParserTest.java:284]: Nice catch :-)

**Comment by dcleao** [comment]
Rebased and fixed the unit test which was lacking assertions.

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 21m 45s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
core,plugins/connections/ui
```
---
:ok_hand: All tests passed!
 
Tests run: **1411**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9455/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**